### DISCUSSION TITLE
Embarcadero C++  11.x

### DISCUSSION THREAD

**Discussion by DaleBarnes42** [magiblot/tvision#105] [Q&A]
Has anyone successfully been able to get this TV to compile under Embarcadero C++ 11x ?  I am really new to C++ and looking for any helpful hints.  Not looking for be given the package that works (unless you want to share) but just sitting here  tinkering and wanting to give this a try.

Thanks again for any hits.

**Comment by magiblot** [comment]
Hi Dale!

Around two years ago I gave Embarcadero's RAD Studio a try. I don't remember if I managed to build the library, and I can't give you exact instructions. However, I do remember that the compiler will be unable to build this TV as long as it defines the `__BORLANDC__` macro.

Given that you say that you are really new to C++, I will skip the technical details. I can only suggest you to use a development enviroment which integrates easily with CMake. I know that Microsoft's Visual Studio supports this; in [this issue's comments](https://github.com/magiblot/tvision/issues/50#issuecomment-808296503) you can find an example of how to create a Turbo Vision application with CMake which you can then load into Visual Studio.

**Comment by DaleBarnes42** [comment]
Great to hear from you.  The reason I was trying to get it to compile with
Embarcadero v11 is I understand if I can get this compiled into a library,
that Delphi 11.x can access it and use it within Delphi.  I have the
ability to compile my pascal packages written with Firemonkey for
Windows/MAC/Linux server.   I have been in talks with Embarcadero to get
the complete source to the pascal version but since yours was already
written and I am told can be called from within Delphi 11 as well.  Thought
it would be pretty sweet to give a try.  Embarcadero may not have all the
source anyway so would be a large project.

I have a  986 app that uses TVision that I thought would be cool if I could
get it updated.

Will keep playing along with it.

**Comment by paule32** [comment]
The C++ Community Edition is free of charge (one year license) the used standard is c++17
